### PR TITLE
[AGDIGGER-143] Add role for git installation

### PR DESCRIPTION
**Pull request by matskiv**
**Why:** we are using git to install homebrew, but git is not available on clean OSX installation. 

JIRA: https://issues.jboss.org/browse/AGDIGGER-143

### CODE REVIEW

File: provision-osx/defaults/main.yml [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 remote_tmp_dir: /tmp
 
+git_install_url: https://netix.dl.sourceforge.net/project/git-osx-installer/git-2.13.1-intel-universal-mavericks.dmg
```
Comment (matskiv) [inline provision-osx/defaults/main.yml]: I am not sure that this is ideal source link.

File: provision-osx/tasks/install_git.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  when: git_installed.rc != 0
+
+-
+  name: Install git package
+  command: "installer -package {{remote_tmp_dir}}/{{git_install_volume_name}}/{{git_install_pkg_name}} -target /"
+  become: yes
+  when: git_installed.rc != 0
+
+-
+  name : Detach git installation dmg
```
Comment (aliok) [inline provision-osx/tasks/install_git.yml]: Maybe delete the DMG file afterwards?

File: provision-osx/defaults/main.yml (hunk 1 above)
Comment (aliok) [inline provision-osx/defaults/main.yml]: I used https://sourceforge.net/projects/git-osx-installer/files/git-2.13.1-intel-universal-mavericks.dmg/download?use_mirror=autoselect and it worked.

of course, you have to rename the output file to something.dmg

File: provision-osx/tasks/install_git.yml (hunk 2 above)
Comment (matskiv) [inline provision-osx/tasks/install_git.yml]: I can surely add that, but by default - files that aren't accessed in three days are deleted from /tmp. And if installation fails down the road due to problems with git, we at least have exact .dmg that was used.

Comment (aliok) [inline provision-osx/tasks/install_git.yml]: ok, then code looks fine. haven't done any verification yet though.

File: provision-osx/tasks/install_git.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+
+-
+  name: Check if git is installed
+  command: git --version
+  failed_when: no
+  register: git_installed
```
Comment (PhilipGough) [inline provision-osx/tasks/install_git.yml]: maybe add `changed_when: False` here also

File: provision-osx/defaults/main.yml [hunk 4]
```diff
@@ -4,6 +4,11 @@
 
 remote_tmp_dir: /tmp
 
+git_install_url: https://sourceforge.net/projects/git-osx-installer/files/git-2.13.1-intel-universal-mavericks.dmg/download?use_mirror=autoselect
```
Comment (PhilipGough) [inline provision-osx/defaults/main.yml]: Taken a look, this URL seems the best choice. May have some issue with productisation though

File: provision-osx/tasks/install_git.yml (hunk 2 above)
Comment (PhilipGough) [inline provision-osx/tasks/install_git.yml]: Yeah I think this is fine @aliok

**Comment by PhilipGough** [comment]
@matskiv changes look good. Have we a clean vm that I could verify this on

**Comment by matskiv** [comment]
Closing as original problem will be resolved by https://issues.jboss.org/browse/AGDIGGER-155

**Pull request closed without merging**